### ISSUE TITLE
Prepare AiiDAlab deployment on Azure with AiiDA 2.x and test it

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida-tutorials#419]
@csadorf @unkcpz @superstar54 @chrisjsewell @mbercx

**Comment by csadorf** [comment]
I have prepared a test deployment at https://aiida-tutorial-2022-test.aiidalab.xyz based on the [revised docker stack](https://github.com/aiidalab/aiidalab-docker-stack) with Python 3.9.4 and AiiDA 2.0.0.

**Comment by mbercx** [comment]
With @csadorf's help, I've [adapted the docker stack](https://github.com/aiidalab/aiidalab-docker-stack/pull/308) to our needs for the tutorial (Installing QE, `aiida-quantumespresso`, ...), and have updated the versions to the latest releases (see email for link). This should be close to the final deployment we want to use for the tutorial. For the testing on Azure, I've made the following assignments based on the presenters of the corresponding sessions in the schedule:

- [x] Basics + Running processes: I've tested these.
- [x] [Managing Data](https://aiida-tutorials.readthedocs.io/en/tutorial-2022-intro/sections/managing_data/index.html): @unkcpz 
- [x] Writing workflows - [Basics](https://aiida-tutorials.readthedocs.io/en/tutorial-2022-intro/sections/writing_workflows/index.html): I'll take care of these (Up to and including Debugging)
- [x] Writing workflows - [Advanced](https://aiida-tutorials.readthedocs.io/en/tutorial-2022-intro/sections/writing_workflows/index.html): @ramirezfranciscof can you test from the Real-world example?
- [ ] [Creating Plugins](https://aiida-tutorials.readthedocs.io/en/tutorial-2022-intro/sections/creating_plugins/index.html): @chrisjsewell

If you're running Quantum ESPRESSO for your sections, you still need to set up the `pw.x` code:

```
verdi code setup --label pw --computer localhost --remote-abs-path /opt/conda/bin/pw.x --input-plugin quantumespresso.pw --non-interactive
```

And install the SSSP efficiency pseudopotential family:

```
aiida-pseudo install sssp
```

Additional Notes:

- Make PRS to the branch: https://github.com/aiidateam/aiida-tutorials/tree/tutorial-2022-intro
- Make sure to also start from the RTD build based on this branch: https://aiida-tutorials.readthedocs.io/en/tutorial-2022-intro/index.html
- Best to set environmental variable: `export AIIDA_WARN_v3=1`, to enable deprecation warnings
- Let me know if you need me to reset your data/server (note that all your data will be lost).
- Double-check: https://github.com/aiidateam/aiida-core/wiki/AiiDA-2.0-plugin-migration-guide
- Look at https://aiida.readthedocs.io/projects/aiida-core/en/latest/reference/_changelog.html#v2-0-0-2022-04-27, and consider if anything new should be added/changed

**Comment by unkcpz** [comment]
It might be an issue that when the first time starts the AiiDAlab server, I encountered a timeout and the server failed to start error. I have to refresh to relaunch the server. If there is an option to increase the timeout might be helpful. @csadorf @mbercx

**Comment by mbercx** [comment]
Thanks @unkcpz for letting us know! I'll adapt the timeout in the terraform setup.

@csadorf just to be sure, the value to be adapted is the `startTimeout` here, right?

```yaml
singleuser:
[...]
  memory:
    limit: 4G
    guarantee: 2G
  cpu:
    limit: 3.0
    guarantee: 1.0
  startTimeout: 300

```

**Comment by csadorf** [comment]
> Thanks @unkcpz for letting us know! I'll adapt the timeout in the terraform setup.
> 
> @csadorf just to be sure, the value to be adapted is the `startTimeout` here, right?

Yes, I believe that should be sufficient. I'm just a bit surprised that this has not been an issue during your tests.

**Comment by mbercx** [comment]
> I'm just a bit surprised that this has not been an issue during your tests.

Yeah, strange that the issue only pops up now. The timeout is currently at 120s, maybe @unkcpz was just unlucky. 😅 

I'll update the deployment later tonight, after I'm done with testing the basic workflows section.

**Comment by ramirezfranciscof** [comment]
>     * [ ]  Writing workflows - [Basics](https://aiida-tutorials.readthedocs.io/en/tutorial-2022-intro/sections/writing_workflows/index.html): I'll take care of these (Up to and including Debugging)
> 
>     * [ ]  Writing workflows - [Advanced](https://aiida-tutorials.readthedocs.io/en/tutorial-2022-intro/sections/writing_workflows/index.html): @ramirezfranciscof can you test from the Real-world example?

Donzo  #439. Actually, I didn't notice the difference you pointed out there between up to debugging / after debuggin, so I just went through all the workflow section, sorry! I hope I didn't overstep on what you were checking @mbercx .
